bootstrap/api/v1beta1/rke2config_types.go

```diff
@@ -147,6 +147,7 @@ type RKE2AgentConfig struct {
 	LoadBalancerPort int `json:"loadBalancerPort,omitempty"`
 
 	// Version specifies the rke2 version.
+	// This field will be deprecated in newer versions of the API and RKE2ControlPlaneSpec.Version will be used instead.
```

Comment:
Could we use the `+kubebuilder:deprecatedversion:warning=""` annotation?

Reply:
Until we resolve the way we manage the new `Version` field ([here](https://github.com/rancher-sandbox/cluster-api-provider-rke2/issues/229)), we're still using this and the other is just a placeholder required by the ClusterClass implementation. The deprecation warning may push people to use the new field while it is technically not supported yet.

Reply:
We could modify the code to use one or the other, and then have the deprecation warning. Initially didn't want to include this in the same PR, but we can do it.

Reply:
I would say lets do it or create an issue to do it straight afterwards so that its in the API bump release.